Harvard students replication.R

```diff
@@ -0,0 +1,349 @@
+
+##Run with PQL.
+ 
+out.pql <- glmmPQL(rexvote ~ male + age1 + age2 + age4 + mye1 + mye2 + farmerown + worker + retired + unemployed + zlrs + euschlecht + zsatisdmo + disp + lfed1 + zasylumseekers + zsur + zasylumseekers:zsur + zreplacementrate + zsur:zreplacementrate + zasylumseekers:zreplacementrate + rmax + salienzmean.c + rvar.c + rvar.c:salienzmean.c + country - 1, random=~1|kontext, family=binomial(link="logit"), data=data, verbose=TRUE)
+
+summary(out.pql)
+
+####Simulations for the Original Model with Centered Variables##
+
+##Simulate betas for centered data set.
+
+set.seed(02138)
+simbetas <- mvrnorm(n=10000, mu=out.pql$coef$fixed, Sigma=out.pql$var)
+
+##Create matrix for all variables.
+
+mat1 <- cbind(data$rexvote, data$male, data$age1, data$age2, data$age4, data$mye1, data$mye2, data$farmerown, 
+data$worker, data$retired, data$unemployed, data$zlrs, data$euschlecht, data$zsatisdmo, data$disp, data$lfed1, 
+data$zasylumseekers, data$zsur, data$zreplacementrate, data$rmax, salienzmean.c, rvar.c, data$at, data$be, data$deo, 
+data$dew, data$dk, data$es, data$fi, data$fr, data$gr, data$it, data$nl, data$no, data$pt, data$se, data$zasylumseekers*data$zsur, 
+data$zsur*data$zreplacementrate, data$zasylumseekers*data$zreplacementrate, salienzmean.c*rvar.c)
+
+##Omit data and create x matrix.
+
+mat2 <- na.omit(mat1)[,-1]
+
+##Create means for all variables
+
+means <- as.vector(apply(mat2, 2, mean))
+
+##Simulate change in prob of ER vote due to change in immigration (asylum seekers) with all other variables held at mean.
+
+asy <- seq(from = -.98, to=4.46, by=.01)
+ests <- matrix(data=NA, ncol=length(asy) ,nrow=10000)
+
+for(j in 1:length(asy)){
+	data.asy <- means
+	data.asy[16] <- asy[j] 
+	data.asy[36] <- asy[j]*data.asy[17]  ##Interaction between asylumseekers and zsur (17).
+	data.asy[38] <- asy[j]*data.asy[18]  ##Interaction between asylumseekers and replacementrate (18). 
+	ests[,j] <- (1+exp(-data.asy%*%t(simbetas)))^-1
+	}
+
+pdf(file="asylumseekers.pdf")
+plot(NA,NA, ylim=c(0,.04), xlim=c(-1,5), xlab="Asylum Seekers (centered)", ylab="Probability of Voting for the ER", main="The Effect of Asylum Seekers on Voting for the ER", cex.main=1)
+lines(asy, apply(ests, 2, mean))
+lines(asy, apply(ests, 2, quantile, 0.025),col="red",lty=3)
+lines(asy, apply(ests, 2, quantile, 0.975),col="red",lty=3)
+abline(h=0)
+dev.off()
+
+
+####################Net Migration Variable########################
+
+##Load migration data.
+
+mig.data <- read.csv("migration.csv")
+
+##Label migration data.
+
+colnames(mig.data) <- c("year","migration","at","be","dk","fi","fr","deo","dew","gr","it","nl","no","pt","es","se","lu","sortcountry")
+
+##Merge migration data with Arzheimer dataset by year and country.
+
+new.data <- merge(data, mig.data, by=c("year","at","be","dk","fi","fr","deo","dew","gr","it","nl","no","pt","es","se","lu","sortcountry"))

```

Comment:
did you try (not sure if this works) using : by = colnames(mig.data). rather than having to rewrite the line of data.